### PR TITLE
Make GlobalError derive Clone

### PR DESCRIPTION
**Pull request by NathanFlurry**

### CODE REVIEW

**Comment by NathanFlurry** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/329?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #324** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/324?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #314** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/314?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #310** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/310?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #311** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/311?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #312** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/312?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #313** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/313?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #315** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/315?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #316** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/316?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **PR #326** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/326?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **PR #325** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/325?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                    * **PR #329** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/329?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/rivet-gg/rivet/329?utm_source=stack-comment).

**Comment by NathanFlurry** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/329?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #324** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/324?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #314** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/314?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #310** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/310?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #311** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/311?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #312** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/312?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #313** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/313?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #315** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/315?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #316** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/316?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **PR #326** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/326?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **PR #325** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/325?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                    * **PR #329** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/329?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/rivet-gg/rivet/329?utm_source=stack-comment).

File: lib/global-error/src/error.rs [hunk 1]
```diff
@@ -8,7 +8,7 @@ use types::rivet::chirp;
 
 pub type GlobalResult<T> = Result<T, GlobalError>;
 
-#[derive(Debug)]
+#[derive(Debug, Clone)]
```
Comment (MasterPtato) [inline lib/global-error/src/error.rs:11]: Why is this needed?

**Review by MasterPtato — CHANGES_REQUESTED** [review]

**Review by AngelOnFira — APPROVED** [review]

File: lib/global-error/src/error.rs (hunk 1 above)
Comment (AngelOnFira) [reply lib/global-error/src/error.rs:11]: Hmm from my perspective, something generic like this probably should implement clone unless there's a reason it shouldn't. Is there something that comes to mind for why it shouldn't?

**Review by MasterPtato — APPROVED** [review]
